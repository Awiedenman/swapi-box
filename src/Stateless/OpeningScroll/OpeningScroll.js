import React from 'react';
import './OpeningScroll.css';
import PropTypes from 'prop-types';

const OpeningScroll = ( {randomScroll} ) => {
   return (
    <div className ="fade">
      <section className="star-wars">
        <div className="crawl">
            <h1 className = "title" > { randomScroll.title} </h1>
            <p>{ randomScroll.crawl }</p>
            <h2>{ randomScroll.releaseDate } </h2>
        </div>
      </section>
    </div>
   )
}


OpeningScroll.propTypes ={
  randomScroll: PropTypes.oneOfType([ PropTypes.array, PropTypes.object ])
}

export default OpeningScroll;