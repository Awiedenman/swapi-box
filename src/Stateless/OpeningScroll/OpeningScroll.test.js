import React from 'react';
import OpeningScroll from './OpeningScroll';
import { shallow } from 'enzyme';

describe( 'OpeningScroll', () => {
  it('should match the snapshot of  scroll on page load', () =>{
    const mockRandomScroll = {
      crawl: "War! The Republic is crumbling",
      releaseDate: "2005-05-19",
      title: "Revenge of the Sith"
    };
    const wrapper = shallow(
      <OpeningScroll 
        randomScroll= { mockRandomScroll }
      />
    );

    expect( wrapper ).toMatchSnapshot();
  });
});
