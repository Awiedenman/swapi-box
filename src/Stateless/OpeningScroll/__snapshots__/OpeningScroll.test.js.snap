// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningScroll should match the snapshot of  scroll on page load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningScroll
    randomScroll={
        Object {
            "crawl": "War! The Republic is crumbling",
            "releaseDate": "2005-05-19",
            "title": "Revenge of the Sith",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="star-wars"
>
        <div
                className="crawl"
        >
                <h1
                        className="title"
                >
                         
                        Revenge of the Sith
                         
                </h1>
                <p>
                        War! The Republic is crumbling
                </p>
                <h2>
                        2005-05-19
                         
                </h2>
        </div>
</section>,
      "className": "fade",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
>
          <h1
                    className="title"
          >
                     
                    Revenge of the Sith
                     
          </h1>
          <p>
                    War! The Republic is crumbling
          </p>
          <h2>
                    2005-05-19
                     
          </h2>
</div>,
        "className": "star-wars",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title"
>
               
              Revenge of the Sith
               
</h1>,
            <p>
              War! The Republic is crumbling
</p>,
            <h2>
              2005-05-19
               
</h2>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Revenge of the Sith",
                " ",
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Revenge of the Sith",
              " ",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "War! The Republic is crumbling",
            },
            "ref": null,
            "rendered": "War! The Republic is crumbling",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2005-05-19",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "2005-05-19",
              " ",
            ],
            "type": "h2",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <h1
                              className="title"
                    >
                               
                              Revenge of the Sith
                               
                    </h1>
                    <p>
                              War! The Republic is crumbling
                    </p>
                    <h2>
                              2005-05-19
                               
                    </h2>
          </div>
</section>,
        "className": "fade",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <h1
                        className="title"
            >
                         
                        Revenge of the Sith
                         
            </h1>
            <p>
                        War! The Republic is crumbling
            </p>
            <h2>
                        2005-05-19
                         
            </h2>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="title"
>
                 
                Revenge of the Sith
                 
</h1>,
              <p>
                War! The Republic is crumbling
</p>,
              <h2>
                2005-05-19
                 
</h2>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Revenge of the Sith",
                  " ",
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Revenge of the Sith",
                " ",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "War! The Republic is crumbling",
              },
              "ref": null,
              "rendered": "War! The Republic is crumbling",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2005-05-19",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "2005-05-19",
                " ",
              ],
              "type": "h2",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
